lib/core/src/receive_swap.rs

```diff
@@ -114,24 +121,38 @@ impl ReceiveSwapHandler {
                     ));
                 }
 
-                // looking for lockup script history to verify lockup was broadcasted
+                // Looking for lockup script history to verify lockup was broadcasted
+                let lockup_tx = match self
+                    .verify_lockup_tx(&receive_swap, &transaction.id, &transaction.hex, false)
+                    .await
+                {
+                    Ok(lockup_tx) => lockup_tx,
+                    Err(e) => {
+                        return Err(anyhow!(
+                        "Swapper mempool reported lockup could not be verified. txid: {}, err: {}",
+                        transaction.id,
+                        e
+                    ));
+                    }
+                };
+
                 if let Err(e) = self
-                    .verify_lockup_tx(&receive_swap, &transaction, false)
+                    .verify_lockup_tx_amount(&receive_swap, &lockup_tx)
                     .await
                 {
+                    // The lockup amount in the tx is underpaid compared to the expected amount
+                    self.update_swap_info(id, Failed, None, None, None, None)?;
```

Comment:
Is there a chance the recoverer will change this back to pending?

Reply:
It won't include the swap at the start of the recover if it's failed, and if this is set to Failed during the recover the `version` update should prevent it